rust/src/operations/merge.rs

```diff
@@ -452,28 +457,71 @@ struct MergeOperation {
 }
 
 impl MergeOperation {
-    pub fn try_new(
-        snapshot: &DeltaTableState,
-        state: &Option<&SessionState>,
+    fn try_from(
+        config: MergeOperationConfig,
+        schema: &DFSchema,
+        state: &SessionState,
+        target_alias: &Option<String>,
+    ) -> DeltaResult<MergeOperation> {
+        let mut ops = HashMap::with_capacity(config.operations.capacity());
+
+        for (column, expression) in config.operations.into_iter() {
+            // Normalize the column name to contain the target alias. If a table reference was provided ensure it's the target.
+            let column = match target_alias {
+                Some(alias) => {
+                    let r = TableReference::bare(alias.to_owned());
+                    match column {
+                        Column {
+                            relation: None,
+                            name,
+                        } => Column {
+                            relation: Some(r),
+                            name,
+                        },
+                        Column {
+                            relation: Some(TableReference::Bare { table }),
+                            name,
+                        } => {
+                            if table.eq(alias) {
+                                Column {
+                                    relation: Some(r),
+                                    name,
+                                }
+                            } else {
+                                return Err(DeltaTableError::Generic(
+                                    "Column must reference column in Delta table".into(),
+                                ));
```

Comment:
Given the feedback from Ion, seems like this might be confusing. Perhaps we should give a message with a hint:

```suggestion
                                return Err(DeltaTableError::Generic(
                                    format!("Table alias '{table}' in column reference '{table}.{name}' unknown. Hint: You must reference the Delta Table with alias '{alias}'.")
                                ));
```